addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/NetatmoBridgeHandler.java

```diff
@@ -0,0 +1,181 @@
+    private static final String HTTP_GET = "GET";
+    private static final String HTTP_POST = "POST";
+    private static final String CHARSET = "UTF-8";
+    private static final String HTTP_CONTENT_TYPE = "application/x-www-form-urlencoded;charset=" + CHARSET;
+    private static final int HTTP_REQUEST_TIMEOUT = 10000;
+    private static final ObjectMapper JSON = new ObjectMapper();
+
+    protected static final Properties HTTP_HEADERS;
+
+    static {
+        HTTP_HEADERS = new Properties();
+        HTTP_HEADERS.put("Accept", "application/json");
+    }
+
+    private static Logger logger = LoggerFactory.getLogger(NetatmoBridgeHandler.class);
+
+    private long refreshInterval;
+    private ScheduledFuture<?> pollingJob;
+
+    public DeviceListBody deviceList;
+    public OAuthCredentials credentials;
+
+    private synchronized void onUpdate() {
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 1, refreshInterval, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private Runnable pollingRunnable = new Runnable() {
+        @Override
+        public void run() {
+            logger.debug("Querying Netatmo API");
+            for (Thing thing : getThing().getThings()) {
+                ThingHandler handler = thing.getHandler();
+                if (handler instanceof AbstractEquipment) {
+                    ((AbstractEquipment) handler).poll();
+                }
+            }
+        }
+    };
+
+    public NetatmoBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Netatmo API bridge handler.");
+
+        NetatmoConfiguration configuration = getConfigAs(NetatmoConfiguration.class);
+
+        refreshInterval = configuration.refreshInterval;
+
+        credentials = new OAuthCredentials(configuration.clientId, configuration.clientSecret,
+                configuration.refreshToken);
+
+        refreshAccessToken();
+
+        deviceList = executeGet(new DeviceListRequest(credentials), NetatmoResponses.DeviceList.class, true);
+
+        updateStatus(ThingStatus.ONLINE);
+        // workaround for issue #92: getHandler() returns NULL after
+        // configuration update. :
+        getThing().setHandler(this);
+
+        onUpdate();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Nothing to do, we're a read only binding
+    }
+
+    /**
+     * Launches an http get request toward Netatmo API
+     *
+     * @param request
+     *            The request to be execute
+     * @param beanClass
+     *            The response class of request
+     * @param retryAuth
+     *            True if the request shall be retried
+     * @return
+     */
+    public <T extends NetatmoResponse<F>, F> F executeGet(AbstractRequest request, Class<T> beanClass,
```

Comment:
@robnielsen thanks for your points, will take them - but if you've got better to spend your time, feel free - I will post a complete new PR, I'm currently refactoring it to integrate Thermostat and hopefully move to gson instead of Jackson.